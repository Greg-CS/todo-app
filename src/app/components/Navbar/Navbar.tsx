"use client"

import React, {useState, useEffect} from 'react'
import Link from 'next/link'
import { usePathname } from 'next/navigation'


export const Navbar = ({user} : {user: any}) => {
  const router = usePathname();
  const [isLoginPage, setIsLoginPage] = useState(false);
  useEffect(() => {
    if (router === '/login') {
      setIsLoginPage(true);
    } else {
      setIsLoginPage(false);
    }
  }, [router]);
    
  return (
  <nav className="navbar bg-gradient-to-t from-[#F7D1CD] via-[#F7D1CD] to-[#B392AC] text-black shadow-2xl h-[10vh]">
    <div className="navbar-start">
      <Link href={"/"} className="btn rounded-2xl bg-[#B392AC] hover:bg-[#D1B3C4] border-0 text-xl text-[#735D78] font-bold">Todo App</Link>
    </div>
    {!user ? 
    <>
      {isLoginPage ? null : (
      <div className='navbar-end gap-3'>
        <Link href={"/login"} className="btn rounded-2xl bg-[#B392AC] hover:bg-[#D1B3C4] border-0 text-xl text-[#735D78] font-bold">Login</Link>
          <span className='text-[#735D78] font-bold'>
            /
          </span>
        <button className="btn rounded-2xl bg-[#B392AC] hover:bg-[#D1B3C4] border-0 text-xl text-[#735D78] font-bold">Sign Up</button>
        {/* <SideMenu/> */}
      </div>
      )}
    </>
    :
    <div className='flex gap-4 navbar-end'>
      <Link href={`/account/${user.id}`}>
        <button className="btn rounded-2xl bg-[#B392AC] hover:bg-[#D1B3C4] border-0 text-xl text-[#735D78] font-bold">App</button>
      </Link>
      <form action="/auth/signout" method="post">
        <button
          className="btn rounded-2xl bg-[#B392AC] hover:bg-[#D1B3C4] border-0 text-xl text-[#735D78] font-bold"
          type="submit"
        >
          Log out
        </button>
      </form>
    </div>
    }
  </nav>
  )
}
