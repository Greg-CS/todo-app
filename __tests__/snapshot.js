import { render } from "@testing-library/react";
import { Navbar } from "../src/app/components/Navbar/Navbar";
it("renders homepage unchanged", () => {
  const { container } = render(<Navbar />);
  expect(container).toMatchSnapshot();
});
