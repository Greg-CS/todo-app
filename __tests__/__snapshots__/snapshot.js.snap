// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <nav
    class="navbar bg-gradient-to-t from-[#F7D1CD] via-[#F7D1CD] to-[#B392AC] text-black shadow-2xl h-[10vh]"
  >
    <div
      class="navbar-start"
    >
      <a
        class="btn rounded-2xl bg-[#B392AC] hover:bg-[#D1B3C4] border-0 text-xl text-[#735D78] font-bold"
        href="/"
      >
        Todo App
      </a>
    </div>
    <div
      class="navbar-end gap-3"
    >
      <a
        class="btn rounded-2xl bg-[#B392AC] hover:bg-[#D1B3C4] border-0 text-xl text-[#735D78] font-bold"
        href="/login#login"
      >
        Login
      </a>
      <span
        class="text-[#735D78] font-bold"
      >
        /
      </span>
      <a
        class="btn rounded-2xl bg-[#B392AC] hover:bg-[#D1B3C4] border-0 text-xl text-[#735D78] font-bold"
        href="/login#signup"
      >
        Sign Up
      </a>
    </div>
  </nav>
</div>
`;
